bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/handler/SerialBridgeThingHandler.java

```diff
@@ -31,31 +37,69 @@
 @NonNullByDefault
 public class SerialBridgeThingHandler extends KNXBridgeBaseThingHandler {
 
-    private final SerialClient client;
+    private @Nullable SerialClient client = null;
+    private @Nullable Future<?> initJob = null;
+
+    private final Logger logger = LoggerFactory.getLogger(SerialBridgeThingHandler.class);
 
     public SerialBridgeThingHandler(Bridge bridge) {
         super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        // create new instance using current configuration settings;
+        // when a parameter change is done from UI, dispose() and initialize() are called
         SerialBridgeConfiguration config = getConfigAs(SerialBridgeConfiguration.class);
         client = new SerialClient(config.getAutoReconnectPeriod(), thing.getUID(),
                 config.getResponseTimeout().intValue(), config.getReadingPause().intValue(),
                 config.getReadRetriesLimit().intValue(), getScheduler(), config.getSerialPort(), config.useCemi(),
                 this);
-    }
 
-    @Override
-    public void initialize() {
         updateStatus(ThingStatus.UNKNOWN);
-        client.initialize();
+        // delay actual initialization, allow for longer runtime of actual initialization
+        initJob = scheduler.submit(() -> {
+            initializeLater();
+        });
```

Comment:
You can simplify like that: `initJob = scheduler.submit(this::initializeLater);`

Reply:
thanks, looks much cleaner now